contracts/utils/math/ModularExponentiation.sol

```diff
@@ -0,0 +1,40 @@
+pragma solidity ^0.8.0;
+
+library ModularExponentiation {
+    function modExp(
+        uint256 b,
+        uint256 e,
+        uint256 m
+    ) internal returns (uint256) {
+        require(m != 0, "ModularExponentiation: Can't calculate for modulus equal to zero");
+
+        uint256 result;
+
+        assembly {
```

Comment:
We want to use as little assembly as possible. I don't think we need any assembly at all. I think we can use `address(5).staticcall(abi.encode(32, 32, 32, b, e, m))`.

Reply:
@frangio noted, will be adding these changes

Reply:
Added the needed changes @frangio